### PR TITLE
Job base classes MSI support

### PR DESCRIPTION
**Pull request by drewgillies**
This is @agr's latest work on the job base types supporting MSIs for storage account access. Tested via other changes as discussed offline.

### CODE REVIEW

**Comment by advay26** [comment]
I've consumed these changes in some validation jobs and tested them in DEV, so it all seems to be working fine.

There are some changes here that directly touch GithubIndexer, Catalog2Registration, and NuGet.Services.AzureSearch (DI changes), and I haven't tested these jobs

**Review by advay26 — APPROVED** [review]

File: src/NuGet.Jobs.Common/StorageAccountExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using NuGetGallery;
+
+namespace NuGet.Jobs
+{
+    public static class StorageAccountHelper
+    {
+        private const string StorageUseManagedIdentityPropertyName = "Storage_UseManagedIdentity";
```
[the remark this answers is not in the document]
Comment (advay26) [reply src/NuGet.Jobs.Common/StorageAccountExtensions.cs:16]: @drewgillies A lot of the other MSI and auth related constants live in `NuGet.Services.Configuration`. That project doesn't exist in Andrei's branch, but it is there in dev and main (something to do with the repo merge I imagine). I was planning on adding these 2 properties as constants there in a later PR, but I've created a PR for that change now: https://github.com/NuGet/NuGetGallery/pull/10154.

That way, when you rebase the feature branch on dev, you can use these constants. I'll move Andrei's code to use those constants too in a later PR with my own changes. It shouldn't affect the consumption of his code, so it should be fine to leave that for later

[the remark this answers is not in the document]
Comment (drewgillies) [reply src/NuGet.Jobs.Common/StorageAccountExtensions.cs:16]: Great--thanks!

**Review by joelverhagen — APPROVED** [review]
LGTM. I do wonder if the `Storage_ManagedIdentityClientId` key is necessary long term. Seems fine if we need it for a intermediate phase.

**Comment by advay26** [comment]
@joelverhagen Once we merge this PR, I'll add a new change allowing us to fall back to the generic 'ClientId' if the Storage-specific ClientId isn't specified: https://github.com/NuGet/NuGetGallery/commit/8213c642a1ba43598bdfb2a670a6e39a11c8ef48. I'll do that soon so that people can just start using the generic 'ManagedIdentityClientId' in their configs instead of using a Storage one. And we could just get rid of that Storage constant then if needed.

**Review by agr — APPROVED** [review]

**Pull request merged**